src/internal.c

```diff
@@ -5255,6 +5259,44 @@ int SetSSL_CTX(WOLFSSL* ssl, WOLFSSL_CTX* ctx, int writeDup)
 
     ssl->ctx     = ctx; /* only for passing to calls, options could change */
     ssl->version = ctx->method->version;
+#if defined(OPENSSL_EXTRA) || defined(WOLFSSL_WPAS_SMALL)
+    ssl->options.mask = ctx->mask;
+#endif
+#ifdef OPENSSL_EXTRA
+    if (ssl->version.minor == TLSv1_3_MINOR &&
+     (ssl->options.mask & SSL_OP_NO_TLSv1_3) == SSL_OP_NO_TLSv1_3) {
+        WOLFSSL_MSG("\tOption set to not allow TLSv1.3, Downgrading");
+        ssl->version.minor = TLSv1_2_MINOR;
```

Comment:
And only downgrade if downgrade is on and downgrade version is allowed?

Reply:
The reason why this shouldn't be a hard error is that if a CTX object is created using `wolfTLS_client_method_ex` then this will return the highest supported protocol version. libest then sets the maximum protocol version using `SSL_CTX_set_max_proto_version`. [This behaviour can be seen used here.](https://github.com/julek-wolfssl/libest/blob/74e1d0bb0f5a64d9be8b9d1dca1dc448cf5aab16/src/est/est_client.c#L810)

I added checking if downgrading is allowed.